### ISSUE TITLE
MGS2: Substance(GOG) - All textures are missing (v2.71.2)

### ISSUE THREAD

**Issue by djc5166** [dege-diosg/dgVoodoo2#12]
Removed GOG's d3d8.dll, added dgVoodoo2's. Tried different settings and output API's. Textures always missing.

![image](https://user-images.githubusercontent.com/19207074/96481150-2930c600-1209-11eb-876c-2b828dc6d15e.png)
![image](https://user-images.githubusercontent.com/19207074/96481260-4d8ca280-1209-11eb-9b02-89fd80c342d3.png)

**Comment by dege-diosg** [comment]
Ok, I checked this out. By default I get a black screen, or I'd say, almost, because graphical parts with solid colors show up. So, the problem is the missing textures.
With dgVoodoo it works better but still, some textures are missing.

I don't know what the difference is compared to the original version because that one works perfectly fine.

**Comment by djc5166** [comment]
My black screen on the GOG wrapper was fixed by disabling RTSS hooks for mgs2_sse.exe, guess it was causing problems somehow. Now it seems to work just fine with the GOG wrapper.

Still no textures with dgVoodoo2 though. But at least its usable for me with GOG's solution.

**Comment by dege-diosg** [comment]
Indeed. Speaking about the original version, I found a bug in dgVoodoo which causes the game to work on nVidia but not on Intel HD Graphics (interesting driver implementation difference at low shader-level).

But: the game is also broken with MS D3D8 (Win 10) in the same way. D3D8 and 9 are very similar but aren't exactly the same, there are minor differences in the behavior of these API's.
AFAIK, with appearing Win Vista, MS mapped the D3D8 frontend to the D3D9 DDI (driver backend) and that's what broke this game. Missing textures are due to such an incompatibility between D3D8/9.
GOG however seems to have modified the game executable in favor of D3D9, I can see the difference when debugging the part in question. I won't modify dgVoodoo to expect this behavior because I want to remain compliant to the original MS D3D8.

So, summarized: dgVoodoo won't be compatible with the GOG version but (aside from the bug I encountered) it works with the original.
